frontend/public/components/build-logs.jsx

```diff
@@ -27,25 +36,31 @@ export class BuildLogs extends React.Component {
     };
   }
 
-  static getDerivedStateFromProps(nextProps, prevState) {
-    const phase = _.get(nextProps.obj, 'status.phase');
+  static getDerivedStateFromProps({ obj: build }, { status: prevStatus }) {
+    const phase = _.get(build, 'status.phase');
     const status = buildToLogSourceStatus(phase);
-    if (prevState.status !== status){
+    if (prevStatus !== status){
       return {status};
     }
     return null;
   }
 
   render() {
-    const namespace = _.get(this.props.obj, 'metadata.namespace');
-    const buildName = _.get(this.props.obj, 'metadata.name');
+    const { obj: build } = this.props;
+    const namespace = _.get(build, 'metadata.namespace');
+    const buildName = _.get(build, 'metadata.name');
```

Comment:
I know that it's not new, but we can put those two lines into one since the `name` is not reserved in this scope:
`const { name, namespace } = build.metadata;`